plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsReader.java

```diff
@@ -113,7 +117,7 @@ private static TableStatistics makeTableStatistics(
         }
         long snapshotId = snapshot.get();
 
-        TupleDomain<IcebergColumnHandle> enforcedPredicate = enforcedConstraint;
+        TupleDomain<IcebergColumnHandle> enforcedPredicate = enforcedConstraint.intersect(unenforcedConstraint);
```

Comment:
... and in the second commit the enforcedPredicate is ill-named, because it's not actually enforced

- let's call it `effectivePredicate`.
- let's introduce the variable in the second commit